### PR TITLE
add RobotMeasurementsDeducer

### PR DESCRIPTION
**Pull request by romni**

### CODE REVIEW

File: src/main/java/io/leonis/subra/game/engine/RobotMeasurementsDeducer.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  @Override
+  public Publisher<Robot.Measurements> apply(
+      final Publisher<DatagramPacket> datagramPacketPublisher
+  ) {
+    return Flux.from(datagramPacketPublisher)
+        .map(datagramPacket ->
+            Arrays.copyOfRange(datagramPacket.getData(), 0, datagramPacket.getLength()))
+        .map(LambdaExceptions.rethrowFunction(Robot.Measurements::parseFrom))
+        // get rid of empty measurements
+        .filter(measurementsList -> !measurementsList.getMeasurementsList().isEmpty());
```
Comment (thumbnail) [inline src/main/java/io/leonis/subra/game/engine/RobotMeasurementsDeducer.java]: Do we want a protobuf-agnostic data container for the measurements, or not?

Comment (romni) [reply src/main/java/io/leonis/subra/game/engine/RobotMeasurementsDeducer.java]: could be converted to a `Map<String, Float>`

Comment (thumbnail) [reply src/main/java/io/leonis/subra/game/engine/RobotMeasurementsDeducer.java]: I think that's fine

**Review by thumbnail — CHANGES_REQUESTED** [review]
We lose the id of the robot during the transformation in this way. And i don't know if it's desirable to add it as a key in the measurements map.

**Comment by thumbnail** [comment]
Maybe we can return a `Entry<PlayerIdentity, Map<String, Double>>`. but i think a custom valuemodel might be as good of a solution. (and just return a `PlayerMeasurement`, which is easier to transform as well :)).

**Review by thumbnail — APPROVED** [review]
Pretty cool stuff!

**Pull request merged**